src/components/catalog/PluginCatalog.tsx

```diff
@@ -65,6 +70,17 @@ const PluginCatalog = () => {
     });
   };
 
+  if (loading) {
+    return <>
+      <EmptyState>
+        <EmptyStateIcon variant="container" component={Spinner} />
+        <Title headingLevel='h4'>
```

Comment:
@PintoGideon @mightyFZeus I think an alternative is to make a reusable component called `Loading` that will be used when the loading state is true instead of repeating this code

Reply:
Moreover, this code is not related to this PR so it should not be included in the patch.